File: calendar_Java/JAVA Source Code/src/hpn/settings/LocalNet.java
package hpn.settings;

import java.net.InetAddress;
import java.net.UnknownHostException;

public final class LocalNet
{
    private static String ipAddress;

    public static String IpAddress()
    {
            ipAddress = LocalIPAddress();
            return ipAddress;
    }

    private static String LocalIPAddress()
    {
    	String localIP = "";
    	try {
			InetAddress thisHostIPv4 = InetAddress.getLocalHost();
			localIP = thisHostIPv4.getHostAddress();
		} catch (UnknownHostException e) {
			
		}
        return localIP;
    }
}

File: calendar_C#/CS Source Code/HpnCalendarTools/hpn/settings/LocalNet.cs
using System;
using System.Net;
using System.Net.Sockets;

namespace hpn.settings
{
    public static class LocalNet
    {
        private static String ipAddress;

        public static String IpAddress
        {
            get
            {
                ipAddress = LocalIPAddress();
                return ipAddress;
            }
            set
            {
                ipAddress = LocalIPAddress();
            }
        }

        private static String LocalIPAddress()
        {
            IPHostEntry host;
            string localIP = "";
            host = Dns.GetHostEntry(Dns.GetHostName());
            foreach (IPAddress ip in host.AddressList)
            {
                if (ip.AddressFamily == AddressFamily.InterNetwork)
                {
                    localIP = ip.ToString();
                    break;
                }
            }
            return localIP;
        }
    }
}
